solr/core/src/java/org/apache/solr/pkg/PackageListeners.java

```diff
@@ -63,13 +63,13 @@ public synchronized void removeListener(Listener listener) {
   }
 
   synchronized void packagesUpdated(List<PackageLoader.Package> pkgs) {
-    MDCLoggingContext.setCore(core);
```

Comment:
Why do we need this check? setCore already handles nulls.

Reply:
because we are calling a 

`MDCLoggingContext.clear();` in finally

Reply:
I was under the impression that it setCore/clear still work even if the core was null.

Reply:
OK, got it. I misunderstood the comment